AVInput/AVInputImplementation.cpp

```diff
@@ -0,0 +1,1949 @@
+    }
+
+    Core::hresult AVInputImplementation::GetSupportedGameFeatures(IStringIterator*& features)
+    {
+        Core::hresult result = Core::ERROR_NONE;
+        features = nullptr;
+        std::vector<std::string> supportedFeatures;
+        try {
+            device::HdmiInput::getInstance().getSupportedGameFeatures(supportedFeatures);
+        } catch (const device::Exception& err) {
+            LOG_DEVICE_EXCEPTION0();
+            result = Core::ERROR_GENERAL;
+        }
+
+        if (!supportedFeatures.empty() && result == Core::ERROR_NONE) {
+            features = Core::Service<RPC::IteratorType<IStringIterator>>::Create<IStringIterator>(supportedFeatures);
+        } else {
+            result = Core::ERROR_GENERAL;
+        }
+        return result;
+    }
+
+    uint32_t AVInputImplementation::getSupportedGameFeaturesWrapper(const JsonObject& parameters, JsonObject& response)
+    {
+        LOGINFOMETHOD();
+        IStringIterator* features = nullptr;
+        Core::hresult result = GetSupportedGameFeatures(features);
+
+        if (result != Core::ERROR_NONE || features == nullptr) {
+            returnResponse(false);
+        } else {
+            vector<string> supportedFeatures;
+            features->Reset(0);
+            string feature;
+            while (features->Next(feature)) {
+                supportedFeatures.push_back(feature);
+            }
+            features->Release();
+            setResponseArray(response, "supportedGameFeatures", supportedFeatures);
+            returnResponse(true);
+        }
+    }
+
+    uint32_t AVInputImplementation::getGameFeatureStatusWrapper(const JsonObject& parameters, JsonObject& response)
+    {
+        LOGINFOMETHOD();
+
+        if (!(parameters.HasLabel("portId") && parameters.HasLabel("gameFeature"))) {
+            LOGWARN("Required parameters are not passed");
+            returnResponse(false);
+        }
+
+        string sPortId = parameters["portId"].String();
+        string sGameFeature = parameters["gameFeature"].String();
+        int portId = 0;
+        try {
+            portId = stoi(sPortId);
+        } catch (...) {
+            LOGWARN("Invalid Arguments");
+            returnResponse(false);
+        }
+
+        bool mode = false;
+        Core::hresult ret = GetGameFeatureStatus(portId, sGameFeature, mode);
+        if (ret == Core::ERROR_NONE) {
+            response["mode"] = mode;
+            returnResponse(true);
+        } else {
+            LOGWARN("AVInputImplementation::getGameFeatureStatusWrapper Mode is not supported. Supported mode: ALLM, VRR-HDMI, VRR-FREESYNC, VRR-FREESYNC-PREMIUM, VRR-FREESYNC-PREMIUM-PRO");
+            returnResponse(false);
+        }
+    }
+
+    Core::hresult AVInputImplementation::GetGameFeatureStatus(int id, const string& feature, bool& mode)
+    {
+        // TODO: The current docs state that the id parameter is optional, but that's not the case in existing code. Which is correct?
```

Comment:
Here's a case where documentation states parameter is optional but not handled as such in code. Which is correct?